src/js/containers/account/visualizations/AccountTimeVisualizationContainer.jsx

```diff
@@ -79,13 +79,85 @@ export class AccountTimeVisualizationSectionContainer extends React.Component {
 
         const searchOperation = new AccountSearchBalanceOperation(this.props.account.id);
         searchOperation.fromState(this.props.reduxFilters);
-        const filters = searchOperation.toParams();
+        // const filters = searchOperation.toParams();
+        const balanceFilters = searchOperation.toParams();
+        let filters = balanceFilters;
 
         const requests = [];
         const promises = [];
 
-        if (this.state.hasFilteredObligated) {
-            // In progress
+        const hasFilteredObligated = (this.props.reduxFilters.objectClass.count() > 0)
```

Comment:
Made hasFilteredObligated a state variable & passed down to child components as a prop